### PR TITLE
Adding signboard info to data processing and triplegbuilder

### PR DESCRIPTION
**Pull request by kdiluca**
# Issue

Added signboards to guidance views

## Tasklist

 - [x] Add tests
 - [ ] Add #fixes with the issue number that this PR addresses
 - [ ] Generally use squash merge to rebase and clean comments before merging
 - [x] Update the [changelog](CHANGELOG.md)

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

File: src/odin/maneuversbuilder.cc [hunk 1]
```diff
@@ -3008,7 +3013,7 @@ void ManeuversBuilder::MatchGuidanceViewJunctions(Maneuver& maneuver,
             (base_prefix == overlay_tokens.at(0))) {
           DirectionsLeg_GuidanceView guidance_view;
           guidance_view.set_data_id(std::to_string(trip_path_->osm_changeset()));
-          guidance_view.set_type("jct"); // TODO implement for real in the future based on sign type
+          guidance_view.set_type("jct");
```
Comment (dgearhart) [inline src/odin/maneuversbuilder.cc:3016]: setup a map with the enum type values to strings

File: src/odin/maneuversbuilder.cc [hunk 2]
```diff
@@ -3019,6 +3024,31 @@ void ManeuversBuilder::MatchGuidanceViewJunctions(Maneuver& maneuver,
...
+    auto curr_edge = trip_path_->GetCurrEdge(node_index);
+    if (curr_edge && (curr_edge->has_sign())) {
+      // Process overlay guidance view signboards
+      for (const auto& base_guidance_view_signboard : curr_edge->sign().guidance_view_signboards()) {
+        auto base_tokens = split(base_guidance_view_signboard.text(), ';');
+        // If base(is_route_number) guidance view board and a pair...
+        if (base_guidance_view_signboard.is_route_number() && is_pair(base_tokens)) {
+          DirectionsLeg_GuidanceView guidance_view;
+          guidance_view.set_data_id(std::to_string(trip_path_->osm_changeset()));
+          guidance_view.set_type("signboard");
```
Comment (dgearhart) [inline src/odin/maneuversbuilder.cc:3043]: same as above

File: valhalla/odin/maneuversbuilder.h [hunk 3]
```diff
@@ -246,6 +246,13 @@ class ManeuversBuilder {
                                   const std::string& base_prefix,
                                   const std::string& base_suffix);
 
+  /**
+   * Process the guidance view signboards.
+   *
+   * @param maneuver The maneuver to process.
+   */
+  void ProcessGuidanceViewSignBoards(Maneuver& maneuver);
```
Comment (dgearhart) [inline valhalla/odin/maneuversbuilder.h:254]: lowercase `b` since it is one word

**Review by dgearhart — CHANGES_REQUESTED** [review]
i left some notes

File: src/odin/maneuversbuilder.cc [hunk 4]
```diff
@@ -3019,6 +3024,31 @@ void ManeuversBuilder::MatchGuidanceViewJunctions(Maneuver& maneuver,
...
+       ((node_index < maneuver.end_node_index()) && (edge_count < kOverlaySignBoardEdgeMax));
+       ++node_index, edge_count++) {
+    // Loop over guidance view signboards
+    auto curr_edge = trip_path_->GetCurrEdge(node_index);
+    if (curr_edge && (curr_edge->has_sign())) {
+      // Process overlay guidance view signboards
+      for (const auto& base_guidance_view_signboard : curr_edge->sign().guidance_view_signboards()) {
+        auto base_tokens = split(base_guidance_view_signboard.text(), ';');
+        // If base(is_route_number) guidance view board and a pair...
+        if (base_guidance_view_signboard.is_route_number() && is_pair(base_tokens)) {
```
Comment (danpaz) [inline src/odin/maneuversbuilder.cc:3040]: Is this check needed given that signboards never have overlays?

File: test/gurka/test_guidance_views_signposts.cc [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+using namespace valhalla;
+
+class GuidanceViews_Signboards : public ::testing::Test {
+protected:
+  static gurka::map map;
+
+  static void SetUpTestSuite() {
+    constexpr double gridsize = 100;
+
+    // A--B-BASE-C--D-OVERLAY-E--F
```
Comment (danpaz) [inline test/gurka/test_guidance_views_signposts.cc:17]: Remove copy paste comment

File: test/gurka/test_guidance_views_signposts.cc [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    map = gurka::buildtiles(layout, ways, nodes, {}, "test/data/guidance_views_signboards", {});
+  }
+};
+
+gurka::map GuidanceViews_Signboards::map = {};
+
+/*************************************************************/
+
+TEST_F(GuidanceViews_Signboards, CheckGuidanceViews) {
+  auto result = gurka::route(map, "A", "G", "auto");
```
Comment (danpaz) [inline test/gurka/test_guidance_views_signposts.cc:48]: Add a test from A->X and check that guidance views are not added

File: src/odin/maneuversbuilder.cc (hunk 4 above)
Comment (gknisely) [reply src/odin/maneuversbuilder.cc:3040]: yes.  base has a route number and there should be a pair

File: proto/directions.proto [hunk 7]
```diff
@@ -68,8 +68,19 @@ message DirectionsLeg {
...
+      kAftertoll = 3;
+      kEnt = 4;
+      kExit = 5;
+      kCityreal = 6;
+      kDirectionboard = 7;
+      kSignboard = 8;
+    }
     optional string data_id = 1;        // TODO future enum as data id?
-    optional string type = 2;           // The type of guidance view TODO future enum
+    optional string type = 2;           // The type of guidance view
```
Comment (dgearhart) [inline proto/directions.proto:83]: `type` should be the enum

File: src/odin/maneuversbuilder.cc [hunk 8]
```diff
@@ -3008,7 +3028,10 @@ void ManeuversBuilder::MatchGuidanceViewJunctions(Maneuver& maneuver,
             (base_prefix == overlay_tokens.at(0))) {
           DirectionsLeg_GuidanceView guidance_view;
           guidance_view.set_data_id(std::to_string(trip_path_->osm_changeset()));
-          guidance_view.set_type("jct"); // TODO implement for real in the future based on sign type
+          guidance_view.set_type(
+              guidanceview_type_string
```
Comment (dgearhart) [inline src/odin/maneuversbuilder.cc:3032]: just assign the enum here and then translate to string in `route_serializer_osrm.cc`

File: src/odin/maneuversbuilder.cc [hunk 9]
```diff
@@ -3019,6 +3042,34 @@ void ManeuversBuilder::MatchGuidanceViewJunctions(Maneuver& maneuver,
...
+    auto curr_edge = trip_path_->GetCurrEdge(node_index);
+    if (curr_edge && (curr_edge->has_sign())) {
+      // Process overlay guidance view signboards
+      for (const auto& base_guidance_view_signboard : curr_edge->sign().guidance_view_signboards()) {
+        auto base_tokens = split(base_guidance_view_signboard.text(), ';');
+        // If base(is_route_number) guidance view board and a pair...
+        if (base_guidance_view_signboard.is_route_number() && is_pair(base_tokens)) {
+          DirectionsLeg_GuidanceView guidance_view;
+          guidance_view.set_data_id(std::to_string(trip_path_->osm_changeset()));
+          guidance_view.set_type(
```
Comment (dgearhart) [inline src/odin/maneuversbuilder.cc:3061]: just assign the enum here and then translate to string in route_serializer_osrm.cc

**Review by gknisely — APPROVED** [review]

**Comment by gknisely** [comment]
@kdiluca looks good but MinGW64 build is failing and out of date with master

**Pull request merged**